_dev/src/router/index.ts

```diff
@@ -81,12 +81,12 @@ const routes: Array<RouteConfig> = [
         component: CampaignList,
       },
       {
-        path: 'creation',
+        path: 'creation/:type',
         name: 'campaign-creation',
         component: CampaignForm,
       },
       {
-        path: 'edit/:id',
+        path: 'edit/:id/:type',
```

Comment:
Hmm. When you edit a campaign, there is no point in providing the type in the route. You should get its type by using the data returned from the API instead.